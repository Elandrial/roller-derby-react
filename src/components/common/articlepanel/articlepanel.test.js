import ArticlePanel from './articlepanel';
import renderer from 'react-test-renderer';

describe('Pager component',() => {
  it('renders correctly', () =>{
    let testData = [{
        "id":2,
        "title":"Integer laoreet ex nec purus posuere",
        "date":"1st April 2022",
        "description":"Nullam maximus efficitur dapibus. Nulla facilisi. Proin mollis enim quis efficitur elementum. Nullam eget gravida quam. Vivamus vel urna vestibulum, aliquam ligula tempor, congue augue.",
        "image": "derby-program.jpg"
    }];
    const tree = renderer
    .create(<ArticlePanel jsondata={testData} ariaName={"Test"} recordsPerPage={3}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  })
});