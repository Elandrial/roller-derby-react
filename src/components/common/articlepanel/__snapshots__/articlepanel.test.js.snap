// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager component renders correctly 1`] = `
Array [
  <article
    className="detail-card"
  >
    <figure>
      <img
        alt="Integer laoreet ex nec purus posuere"
        src="/images/derby-program.jpg"
      />
    </figure>
    <div>
      <h3>
        Integer laoreet ex nec purus posuere
      </h3>
      <time>
        1st April 2022
      </time>
      <p>
        Nullam maximus efficitur dapibus. Nulla facilisi. Proin mollis enim quis efficitur elementum. Nullam eget gravida quam. Vivamus vel urna vestibulum, aliquam ligula tempor, congue augue.
      </p>
      <button
        onClick={[Function]}
      >
        Read More
      </button>
    </div>
  </article>,
  <div
    className="page-control"
  >
    <button
      aria-label="Go to previous Test page"
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      aria-current="true"
      aria-label="Current Test Page, Page 1"
      className=""
      disabled={true}
      onClick={[Function]}
    >
      1
    </button>
    <button
      aria-label="Go to next Test page"
      disabled={true}
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>,
]
`;
